app/views/organizations/adoptable_pets/show.html.erb

```diff
@@ -110,23 +120,13 @@
                         <%= image_tag('green_check.png', height: '21') %>
                       <% end %>
                     </div>
-                  <% else %>
-                    <h4 class="mb-4 mt-3">In love with this pooch?</h4>
-                    <%= form_with model: @adoption_application do |form| %>
-                      <%= form.hidden_field :pet_id, value: @adoption_application.pet_id %>
-                      <%= form.hidden_field :adopter_foster_account_id, value: @adoption_application.adopter_foster_account_id %>
-                      <%= form.submit t(".apply_to_adopt"),
-                        class: "custom-btn-pink",
-                        data: {turbo: false}
-                      %>
-                    <% end %>
+                  <% elsif current_user.adopter_foster_account && current_user.adopter_foster_account.adopter_foster_profile.blank? %>
```

Comment:
@kasugaijin In regards to `is_adopter_or_fosterer?`, if we wanted something like that, it feels like it would fit in the authorizable concern. We could even do something like this to have a predicate method for each role:
```ruby
    ROLES.each do |role|
      define_method "#{role}?" do
        has_role?(role)
      end
    end
```
So instead of having to do `has_role?(:adopter)`, we could do `adopter?`

However, that said, I don't think I like this. If we are sticking with rolify, I think we should use its methods, and this feels like it would just give developers a longer chain to search through to find out what's going on with roles.

I feel the same about `has_adopter_foster_profile?`. `has_blank?` just feels the same as using the Rails built-ins `present?` and `blank?`.

However, one thing that would be nice would be to add a through association to `User` for `adopter_foster_profile`, so that we don't have to always forward through its account. Then we could do:
`current_user.adopter_foster_profile.blank?`

## All that said...
This feels like it's ultimately a question of authorization, and I think all authz should be handled by Action Policy. This conditional:
```ruby
<% elsif current_user.adopter_foster_account && current_user.adopter_foster_account.adopter_foster_profile.blank? %>
```
Can actually be shortened to:
```ruby
allowed_to?(:new?, AdopterFosterProfile)
```
The conditional logic is basically what is happening in the policy already. When we want a conditional to a link, the associated policy should almost always be useful.

I should have caught that that line could also utilize the policy when making the PR. Apologies that I did not suggest it sooner.

Reply:
Excellent suggestion @mononoken , the function ```no_profile?``` in the AdopterFosterPolicy handles the above condition gracefully. Also, it checks for additional constraints on whether the user can create the profile or not. 
Working on the changes 👍

Reply:
Changes done

Reply:
Thanks @mononoken yes on second thoughts I agree with this. We should use what we already have, rather than inject more logic to keep things simpler.

I think this `<% elsif allowed_to?(:new?, AdopterFosterProfile) %>` in place of `<% elsif current_user.adopter_foster_account && current_user.adopter_foster_account.adopter_foster_profile.blank? %>` is better. Though, the one trade off is that it implies the developer understands what the policy is, and does, for it to make sense. So it's not as simple as reading a method that is direct in its naming...but I still think this is fine to use.

Reply:
@kasugaijin I agree that is def one trade off of using the gem, but I think ultimately authz stuff should be handled by the authz tool. And I hope it will be easy for devs to infer the path for `new_adopter_fosterer_profile_path` lines up with `new?, AdopterFosterProfile`. But it does require devs to read the docs for the tool.

As testament to your point though:
```ruby
<% elsif allowed_to?(:create?, with: Organizations::AdopterFosterProfilePolicy, context: {user: current_user}) %>
```
The gem actually does some of this automatically. It could have been shortened to:
```ruby
<% elsif allowed_to?(:new?, AdopterFosterProfile) %>
```
The `user: current_user` context is done automatically by Action Policy as a default. Also, `with:` is unnecessary as it can infer the policy based on the object passed as its second argument (most of the time). However, I think this is good to leave as is. Since the gem is not used much in the views yet, I think more verbosity will be useful for other devs to follow. For example, this more verbosity shows that `allowed_to?` is using the `Policy` objects, and the policy class would lead them to the Action Policy docs. We can always simplify the syntax later.